src/test/java/org/javarosa/core/model/instance/geojson/GeoJsonExternalInstanceTest.java

```diff
@@ -84,6 +95,34 @@ public void parse_addsAllOtherPropertiesAsChildren() throws IOException {
         assertThat(featureCollection.getChildAt(1).getChild("special-property", 0).getValue().getValue(), is("special value"));
     }
 
+    @Test
+    public void parse_usesTopLevelId() throws IOException {
+        TreeElement featureCollection = GeoJsonExternalInstance.parse("id", r("feature-collection-id-toplevel.geojson").toString());
+        assertThat(featureCollection.getChildAt(0).getNumChildren(), is(4));
+        assertThat(featureCollection.getChildAt(0).getChild("id", 0).getValue().getValue(), is("fs87b"));
+
+    }
+
+    @Test
+    public void parse_prioritizesTopLevelId() throws IOException {
+        TreeElement featureCollection = GeoJsonExternalInstance.parse("id", r("feature-collection-id-twice.geojson").toString());
+        assertThat(featureCollection.getChildAt(0).getNumChildren(), is(4));
+        assertThat(featureCollection.getChildAt(0).getChild("id", 0).getValue().getValue(), is("fs87b"));
```

Comment:
Using more test-tailored values here like "top-level-id" instead of a realistic one ("fs87b") would really help the readability of the test given that the JSON lives externally. That goes for the above test as well.

Reply:
Thank you, a nice improvement!